scala-package/core/src/main/scala/org/apache/mxnet/io/NDArrayIter.scala

```diff
@@ -101,20 +102,30 @@ class NDArrayIter(data: IndexedSeq[(String, NDArray)],
   private var cursor = -dataBatchSize
 
   private val (_provideData: ListMap[String, Shape],
-               _provideLabel: ListMap[String, Shape]) = {
+               _provideLabel: ListMap[String, Shape],
+               _provideDataDesc: IndexedSeq[DataDesc],
+               _provideLabelDesc: IndexedSeq[DataDesc]) = {
     val pData = ListMap.empty[String, Shape] ++ initData.map(getShape)
     val pLabel = ListMap.empty[String, Shape] ++ initLabel.map(getShape)
-    (pData, pLabel)
+    val pDData = IndexedSeq.empty[DataDesc] ++ initData.map(ele => {
```

Comment:
do you need derive the DataDesc/LabelDesc when they are passed in the constructor.  Am I missing something?

Reply:
yes, the shape information got update